File: 124_Longest_Consecutive_Sequence.java
public class Solution {
    /**
     * @param nums: A list of integers
     * @return an integer
     */
    public int longestConsecutive(int[] num) {
        // write you code here
        HashSet<Integer> set = new HashSet<>();
        for (int i=0; i<num.length; i++){
            set.add(num[i]);
        }
        
        int longest = 0;
        for (int i=0; i<num.length; i++){
            // look for the upper and lower boundaries of a consecutive
            // sequence from the curret item
            int down = num[i]-1;
            while (set.contains(down)){
                set.remove(down);
                down--;
            }
            int up = num[i]+1;
            while (set.contains(up)){
                set.remove(up);
                up++;
            }
            longest = Math.max(longest, up-down-1);
        }
        return longest;
    }
}

File: 124_Longest_Consecutive_Sequence.py
class Solution:
    """
    @param num, a list of integer
    @return an integer
    """
    def longestConsecutive(self, num):
        # write your code here
        if len(num) == 0:
            return 0;
        numSet = set(num);
        
        longest = 0
        for i in range(len(numSet)):
            left = num[i]-1
            while left in numSet:
                numSet.remove(left)
                left-=1
            right = num[i]+1
            while right in numSet:
                numSet.remove(right)
                right+=1
            longest = max(longest, right-left-1)
        
        return longest
